logstreams/src/main/java/io/zeebe/logstreams/state/StateSnapshotController.java

```diff
@@ -71,28 +81,31 @@ public StateSnapshotMetadata recover(
     }
 
     if (runtimeDirectory.exists()) {
-      controller.delete(runtimeDirectory);
+      FileUtil.deleteFolder(runtimeDirectory.getAbsolutePath());
```

Comment:
RocksDB can write databases in multiple folders; we don't configure it that way at the moment, but we also don't explicitly configure it not that way. That's why deleting and copying a snapshot was done using RocksDB, so that it would reuse the same options file the DB was last written with to properly delete/create the runtime. Doing everything with file ops is maybe slightly faster but potentially breaking.

Reply:
`Files.deleteFolder` should delete recursively so this should be fine anyhow. I would like to leave it as it is. If we have really a problem with this current impl then we can take a look again, but I also not like to open the database and doing an checkpoint, this  sounds weird for me and a bit of misusing the API :sweat_smile: I'm also assume that copying currently not the problem, since this is only done on recovery (startup).

Reply:
On not-small databases however, doing a fully copy is probably more expensive than opening the DB and doing a snapshot (which would copy only a small amount of stuff like the metadata, and hardlink everything else). It also means having always two copies of the database at a time on disk, so if your DB is 2GB, we're effectively using 4GB, etc. 

I admit I'm not sure how big we can expect databases to be, so I'm not sure which one is the better trade off, but I would lean toward having the runtime be a snapshot of a snapshot (i.e. also made up of hardlinks) as being the safer choice.

As for deleting, I won't really insist since at the moment there's not much difference as far as we're concerned (i.e. since we don't configure RocksDB to be split across multiple folders which may or may not be contained in the same parent folder), it just seemed to me like easy future-proofing to delegate it to them (since it's what, 2 calls? `RocksDB.destroyDB` + a `Files.deleteFolder` on the folder for Windows). But I'm fine with just deleting the folder for now; we just need to make sure the DB is not opened when we delete (which is not the case here I think).

Reply:
Lets open an issue for examining this in more detail ok? I would like to complete the current PR. I also saw that Java offers an API to create hardlinks `Files.createLink` but I had then some problems with file permissions. :man_shrugging:

Reply:
created #1812